test/extensions/filters/http/ext_authz/ext_authz_grpc_fuzz_test.cc

```diff
@@ -0,0 +1,128 @@
+#include "envoy/extensions/filters/http/ext_authz/v3/ext_authz.pb.validate.h"
+
+#include "source/common/http/context_impl.h"
+#include "source/common/network/address_impl.h"
+#include "source/extensions/filters/http/ext_authz/ext_authz.h"
+
+#include "test/extensions/filters/common/ext_authz/mocks.h"
+#include "test/extensions/filters/common/ext_authz/test_common.h"
+#include "test/extensions/filters/http/common/fuzz/http_filter_fuzzer.h"
+#include "test/extensions/filters/http/ext_authz/ext_authz_fuzz.pb.validate.h"
+#include "test/extensions/filters/http/ext_authz/ext_authz_fuzz_lib.h"
+#include "test/fuzz/fuzz_runner.h"
+#include "test/mocks/grpc/mocks.h"
+#include "test/mocks/network/mocks.h"
+#include "test/mocks/server/server_factory_context.h"
+#include "test/mocks/upstream/cluster_manager.h"
+
+#include "gmock/gmock.h"
+
+using Envoy::Extensions::Filters::Common::ExtAuthz::TestCommon;
+using testing::Return;
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace ExtAuthz {
+namespace {
+
+std::unique_ptr<envoy::service::auth::v3::CheckResponse>
+makeGrpcCheckResponse(const Grpc::Status::WellKnownGrpcStatus status) {
```

Comment:
Instead of writing code to make CheckResponse, we should change ExtAuthzTest proto
https://github.com/envoyproxy/envoy/blob/412a936100888530cfa1746621a933e78edc78af/test/extensions/filters/http/ext_authz/ext_authz_fuzz.proto#L10

to remove the AuthResult from there, 
https://github.com/envoyproxy/envoy/blob/412a936100888530cfa1746621a933e78edc78af/test/extensions/filters/http/ext_authz/ext_authz_fuzz.proto#L24

instead, adding a CheckResponse message there:
https://github.com/envoyproxy/envoy/blob/412a936100888530cfa1746621a933e78edc78af/api/envoy/service/auth/v3/external_auth.proto#L117

That way, the fuzzer will be able to generate random CheckResponse and feed into the test input:
https://github.com/envoyproxy/envoy/blob/412a936100888530cfa1746621a933e78edc78af/test/extensions/filters/http/ext_authz/ext_authz_fuzz_test.cc#L78

Reply:
A good example like this is the ext_proc fuzzer test input:
https://github.com/envoyproxy/envoy/blob/412a936100888530cfa1746621a933e78edc78af/test/extensions/filters/http/ext_proc/unit_test_fuzz/ext_proc_unit_test_fuzz.proto#L11

Reply:
This is just copied [directly from the old fuzzer code](https://github.com/envoyproxy/envoy/pull/33012/files#diff-de8edc70c81913c4a3089c1e123dd67d2d53860eb3deaa2b6d83e3acc90390cbL28). I will add CheckResponse to the fuzzer input in a subsequent PR.